app/src/main/java/org/openhdp/hdt/ui/tracking/TrackingViewModel.kt

```diff
@@ -24,43 +30,42 @@ class TrackingViewModel @ViewModelInject constructor(
 
     private var job: Job? = null
 
-    fun midnight() = Date().apply {
-        seconds = 0
-        minutes = 0
-        hours = 0
+    fun initialize() {
+        viewModelScope.launch(Dispatchers.Main) {
+            runCatching {
+                val stopwatches = stopwatchRepository.stopwatchDAO.getAllStopwatchesInOrder()
+                val categories = categoryDAO.getAllCategoriesOrdered()
+                stopwatches.mapNotNull { it.asTrackingItem(categories) }
+            }.onFailure {
+                Timber.e(it, "initialize() failure ")
+                _viewState.value = TrackingViewState.Error(it)
+            }.onSuccess { stopwatches ->
+                if (stopwatches.isEmpty()) {
+                    _viewState.value = TrackingViewState.NoStopwatches
+                } else {
+                    _viewState.value = TrackingViewState.Results(ArrayList(stopwatches))
+
+                }
+            }
+        }
+        _viewState.value = TrackingViewState.NoStopwatches
     }
 
-    fun initialize() {
-        _viewState.value = TrackingViewState.Results(
-            arrayListOf(
-                TrackingItem(
-                    "0",
-                    "Netflix",
-                    "",
-                    midnight().time + TimeUnit.HOURS.toMillis(2),
-                    TrackState.INACTIVE,
-                    R.color.colorAccent
-                ),
-                TrackingItem(
-                    "1",
-                    "spacer",
-                    "",
-                    midnight().time + TimeUnit.HOURS.toMillis(1),
-                    TrackState.ACTIVE,
-                    R.color.colorYellow
-                ),
-                TrackingItem(
-                    "2",
-                    "angielski",
-                    "",
-                    midnight().time,
-                    TrackState.ACTIVE,
-                    R.color.colorOrange
-                )
+    private fun Stopwatch.asTrackingItem(categories: List<Category>): TrackingItem? {
+        try {
+            val category = categories.firstOrNull { this.categoryId == it.id } ?: return null
```

Comment:
I think this mapping should be done in DAO as joining Stopwatches and Categories. There is something like this in `StopwatchesWithCategoriesDAO`